python/ray/tests/aws/utils/helpers.py

```diff
@@ -27,3 +32,56 @@ def bootstrap_aws_config(config):
 def bootstrap_aws_example_config_file(file_name):
     config = load_aws_example_config_file(file_name)
     return bootstrap_aws_config(config)
+
+
+def node_provider_tags(config, type_name):
+    """
+    Returns a copy of DEFAULT_NODE_PROVIDER_INSTANCE_TAGS with the Ray node
+    kind and Ray user node type filled in from the input config and node type
+    name.
+
+    Args:
+        config: autoscaler config
+        type_name: node type name
+    Returns:
+        tags: node provider tags
+    """
+    tags = copy.copy(DEFAULT_NODE_PROVIDER_INSTANCE_TAGS)
+    head_name = config["head_node_type"]
+    node_kind = NODE_KIND_HEAD if type_name is head_name else NODE_KIND_WORKER
+    tags[TAG_RAY_NODE_KIND] = node_kind
+    tags[TAG_RAY_USER_NODE_TYPE] = type_name
+    return tags
+
+
+def apply_node_provider_config_updates(config, node_cfg, node_type_name,
+                                       max_count):
+    """
+    Applies default updates made by AWSNodeProvider to node_cfg during node
+    creation.
+
```

Comment:
Can you mention that this is only used in tests?

Reply:
I think this is the only outstanding thing to change here!

Reply:
Updated.